dask/graph_manipulation.py

```diff
@@ -333,11 +383,12 @@ def _bind_one(
         new_layers[new_layer_name], is_bound = layer.clone(
             keys=clone_keys, seed=seed, bind_to=blocker_key
         )
-        new_deps[new_layer_name] = {
+        new_dep = {
             clone_key(dep, seed=seed) for dep in layer_deps_to_clone
         } | layer_deps_to_omit
         if is_bound:
-            new_deps[new_layer_name].add(blocker_key)
+            new_dep.add(blocker_key)
+        new_deps[new_layer_name] = new_dep
```

Comment:
this change is to make mypy happy, as new_deps is a dict of read-only AbstractSets.